### PR TITLE
Add BiGG Metabolite

### PR DESCRIPTION
**Pull request by samuelbunga**
This PR adds the BiGG Metabolite database

### CODE REVIEW

File: src/pyobo/sources/bigg.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    bigg_df = ensure_df(
+        prefix=PREFIX,
+        url=URL,
+        sep="\t",
+        skiprows=18,
+        header=None,
+        names=HEADER,
+        )
+
+    for r, c in bigg_df.iterrows():
```
Comment (cthoyt) [inline src/pyobo/sources/bigg.py:47]: iterating directly over `bigg_df.values` is more efficient. Use in combination with `usecols=...` in the `ensure_df()` function to only get the columns you nedd for iteration

File: src/pyobo/sources/bigg.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+
+"""Converter for bigg."""
+import click
+import bioversions
+from typing import Iterable
+from more_click import verbose_option
+from pyobo.path_utils import ensure_df, ensure_tar_df
```
Comment (cthoyt) [inline src/pyobo/sources/bigg.py:8]: replace this with `pyobo.utils.path`

File: src/pyobo/sources/bigg.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        names=HEADER,
+        )
+
+    for r, c in bigg_df.iterrows():
+        bigg_id = c[0]
+        name = c[2]
+        synonyms = []
+        term = Term(
+            reference=Reference(prefix=PREFIX, identifier=bigg_id, name=name),
+            definition=[],
```
Comment (cthoyt) [inline src/pyobo/sources/bigg.py:53]: invalid type for definition

Comment (cthoyt) [reply src/pyobo/sources/bigg.py:53]: `tox -e mypy` will notify you of stuff like this

**Comment by samuelbunga** [comment]
Hi Charlie, I just added new changes. Please review them in your free time. Thanks!

File: src/pyobo/sources/bigg.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        sep="\t",
+        skiprows=18,
+        header=None,
+        names=HEADER,
+        usecols=['bigg_id', 'name'],
+        force=force,
+        version=version,
+    )
+
+    for v in bigg_df.values:
```
Comment (cthoyt) [inline src/pyobo/sources/bigg.py:51]: `v` is a bad variable name. Better is to use tuple unpacking

```suggestion
    for bigg_id, name in bigg_df.values:
```

File: src/pyobo/sources/bigg.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+def get_terms(force: bool = False, version: Optional[str] = None) -> Iterable[Term]:
+    bigg_df = ensure_df(
+        prefix=PREFIX,
+        url=URL,
+        sep="\t",
+        skiprows=18,
+        header=None,
+        names=HEADER,
+        usecols=['bigg_id', 'name'],
```
Comment (cthoyt) [inline src/pyobo/sources/bigg.py:46]: what about the other columns? `database_links` has lots of information worth parsing out, for example. `old_bigg_ids` can be put in the alternative identifier field in the `pyobo.Term`

File: src/pyobo/sources/bigg.py [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        iter_terms=get_terms,
+        iter_terms_kwargs=dict(force=force, version=version),
+        typedefs=[has_role],
+        synonym_typedefs=[alias_type],
+        auto_generated_by=f"bio2obo:{PREFIX}",
+        data_version=version,
+    )
+
+
+def get_terms(force: bool = False, version: Optional[str] = None) -> Iterable[Term]:
```
Comment (cthoyt) [inline src/pyobo/sources/bigg.py:38]: check CI - this needs a docstring

**Review by cthoyt — CHANGES_REQUESTED** [review]

File: src/pyobo/sources/bigg.py [hunk 7]
```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+
+"""Converter for bigg."""
```
Comment (cthoyt) [inline src/pyobo/sources/bigg.py:3]: fix stylization of BiGG

File: src/pyobo/sources/bigg.py [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+PREFIX = "bigg.metabolite"
+
+URL = "http://bigg.ucsd.edu/static/namespace/bigg_models_metabolites.txt"
+
+alias_type = SynonymTypeDef(id="alias", name="alias")
+has_role = TypeDef(reference=Reference(prefix="bigg", identifier="has_role"))
+
+
+def get_obo(force: bool = False) -> Obo:
+    """Get bigg as OBO."""
```
Comment (cthoyt) [inline src/pyobo/sources/bigg.py:23]: stylization

File: src/pyobo/sources/bigg.py [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+    version = bioversions.get_version("bigg")
+    # version = '1.2'
```
Comment (cthoyt) [inline src/pyobo/sources/bigg.py:25]: remove dead code

File: src/pyobo/sources/bigg.py [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+    return Obo(
+        ontology=PREFIX,
+        name="bigg models metabolites database",
```
Comment (cthoyt) [inline src/pyobo/sources/bigg.py:28]: title case + stylization

File: src/pyobo/sources/bigg.py [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+        iter_terms=get_terms,
+        iter_terms_kwargs=dict(force=force, version=version),
+        typedefs=[has_role],
```
Comment (cthoyt) [inline src/pyobo/sources/bigg.py:31]: why is this included?

File: src/pyobo/sources/bigg.py [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+        names=HEADER,
+        usecols=["bigg_id", "name"],
+        force=force,
+        version=version,
+    )
+
+    for v in bigg_df.values:
+        bigg_id = v[0]
+        name = v[1]
+        synonyms = []
```
Comment (cthoyt) [inline src/pyobo/sources/bigg.py:54]: why include this blank list?

**Review by cthoyt — CHANGES_REQUESTED** [review]

**Pull request merged**